homeassistant/components/wallbox/sensor.py

```diff
@@ -47,91 +45,99 @@
 CONF_STATION = "station"
 UPDATE_INTERVAL = 30
 
-_LOGGER = logging.getLogger(__name__)
+
+@dataclass
+class WallboxRequiredKeysMixin:
+    """Mixin for required keys."""
+
+    value_fn: Callable[[WallboxData], int | float | str]
 
 
 @dataclass
-class WallboxSensorEntityDescription(SensorEntityDescription):
+class WallboxSensorEntityDescription(SensorEntityDescription, WallboxRequiredKeysMixin):
     """Describes Wallbox sensor entity."""
 
-    precision: int | None = None
-
 
 SENSOR_TYPES: dict[str, WallboxSensorEntityDescription] = {
     CONF_CHARGING_POWER_KEY: WallboxSensorEntityDescription(
         key=CONF_CHARGING_POWER_KEY,
         name="Charging Power",
-        precision=2,
         native_unit_of_measurement=POWER_KILO_WATT,
         device_class=DEVICE_CLASS_POWER,
         state_class=STATE_CLASS_MEASUREMENT,
+        value_fn=lambda data: round(data[CONF_CHARGING_POWER_KEY], 2),
```

Comment:
in the SENSOR_TYPES dict we now end up with a lot of repetitiveness for this function -> not really a big fan of that.